code/API_definitions/Edge-Application-Management.yaml

```diff
@@ -1024,33 +1249,23 @@ components:
       type: integer
       description: Port to stablish the connection
       minimum: 0
+
     RequiredResources:
       description: |
         Fundamental hardware requirements to be provisioned by the
         Application Provider.
-      type: object
-      required:
-        - numCPU
-        - memory
-        - storage
-      properties:
-        numCPU:
-          type: integer
-          description: Number of virtual CPUs
-          example: 1
-        memory:
-          type: integer
-          example: 10
-          description: Memory in giga bytes
-        storage:
-          type: integer
-          example: 60
-          description: Storage in giga bytes
-        gpu:
-          type: array
-          description: Number of GPUs
-          items:
-            $ref: '#/components/schemas/GpuInfo'
+      type: array
+      items:
+        oneOf:
+        - $ref: "#/components/schemas/Kubernetes"
```

Comment:
The idea is to manage applications as self-contained units, including the resources they need. This way, the application itself collects all the resources required to work properly. If a second application needs to be deployed on the same cluster, it might indicate that the resource requirements for the first application were overestimated. Here are two options for the developer:

a) Modify the Helm chart to add the application there (even modify the resources to fit - this would require an API for application Update) or
b) Deploy a new application.